containers_test.go

```diff
@@ -201,3 +202,78 @@ func TestUpload(t *testing.T) {
 		t.Errorf("%s not found", path)
 	}
 }
+
+func TestSquashImage(t *testing.T) {
+	client := DockerClient()
+	if client == nil {
+		t.Skip("Could not find Docker daemon connection")
+	}
+
+	repo := "yourbase-layer-test"
+	tag := "v1"
+	image := fmt.Sprintf("%s:%s", repo, tag)
+
+	buildLayeredImage(t, image)
+
+	err := PullImageIfNotHere(ContainerDefinition{
+		Image: image,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	squashImageId, err := imageId(repo, tag)
+	if err != nil {
+		t.Error(err)
+	}
+
+	layers, _ := CountLayersInImage(squashImageId)
+	if layers <= 1 {
+		t.Error("yourbase-layer-test should have more than one layer.")
+	}
+
+	t.Logf("Pre layer count: %d", layers)
+
+	err = SquashImage(context.Background(), repo, tag)
+	if err != nil {
+		t.Errorf("SquashImage failed: %v", err)
+	}
+
+	newImageId, err := imageId(repo, tag)
+	if err != nil {
+		t.Errorf("Couldn't find squashed image: %s", image)
+	}
+
+	// The new image should have only one layer
+	layers, err = CountLayersInImage(newImageId)
+	if err == nil && layers != 1 {
+		t.Error("yourbase-layer-test should be comprised of a single layer.")
+	}
+
+	t.Logf("Post layer count: %d", layers)
+
+	// Clean up
+	if err := client.RemoveImage(newImageId); err != nil {
+		t.Errorf("Could not removed squashed image")
+	}
+}
+
+func buildLayeredImage(t *testing.T, imageName string) {
```

Comment:
In general, it's better to avoid passing around a `*testing.T` because it obfuscates control flow in the containing test. The key constraint is that `Fatal` can't be used outside the test function's goroutine.

Instead, prefer returning an `error` from the function. It composes better and keeps test code looking closer to production code.